modin/data_management/functions/groupby_function.py

```diff
@@ -78,7 +78,7 @@ def reduce(
         **kwargs,
     ):
         by_part = list(df.index.names)
-        if drop and len(df.columns.intersection(by_part)) > 0:
+        if drop and len(df.columns.intersection(pandas.Index(by_part))) > 0:
```

Comment:
This change should not be needed, I have tested it locally.

Reply:
The behavior for `MultiIndex.intersection` was changed for some reason, so now it's raising a TypeError when trying to find an intersection between multi-index and a list.
```python
import pandas

multi_index = pandas.MultiIndex.from_arrays([["a", "b"], ["c", "d"]])
try:
    result = multi_index.intersection(["some_col"])
except Exception as e:
    result = e
print(f"Intersection with a list: {repr(result)}")

result = multi_index.intersection(pandas.Index(["some_col"]))
print(f"Intersection with an Index: {result}")
```

<details><summary>Output for pandas 1.1.5</summary>

```
Intersection with a list: MultiIndex([], )
Intersection with an Index: MultiIndex([], )
```

</details>

<details><summary>Output for pandas 1.2.1</summary>

```
Intersection with a list: TypeError('other must be a MultiIndex or a list of tuples')
Intersection with an Index: MultiIndex([], )
```

</details>

Reply:
I see, let's leave a comment above this then and report the regression to pandas. Leave the issue number in the comment

Reply:
done it